### PR TITLE
I11- unit tests for piece and piece placement

### PR DESCRIPTION
**Pull request by why-xn**

### CODE REVIEW

File: tests/backend/test_GameManager.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+def game_manager():
+    game_manager = GameManager(7, 9)
+    yield game_manager
+
+class TestGameManager:
+    def test_initial_turn(self, game_manager):
+        # Test if the initial turn is WHITE
+        assert game_manager.get_turn() == 'white'
+
+    def test_all_pieces_placed(self, game_manager):
```
Comment (EXC3ll3NTrhyTHM) [inline tests/backend/test_GameManager.py:15]: This test is only testing the negative path for the `placement_complete` function. We should have the happy path test for if `self.pieces.all_pieces_placed()` is true.

File: tests/backend/test_GameManager.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        # Test at initialization all pieces are not expected to be placed
+        assert not game_manager.placement_complete()
+
+    def test_piece_placement(self, game_manager):
```
Comment (EXC3ll3NTrhyTHM) [inline tests/backend/test_GameManager.py:19]: We should also evaluate that the piece exist at expected coordinates.

File: tests/backend/test_GameManager.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        # Test piece placement
+        res = game_manager.piece_placement(0, 0)
+        assert res == 1
+
+        game_manager.end_turn()
+        res = game_manager.piece_placement(0, 0)
+        assert res == "GameManagerError -- invalid piece placement position"
```
Comment (EXC3ll3NTrhyTHM) [inline tests/backend/test_GameManager.py:26]: From the business logic perspective this should be an exception not a string,

File: tests/backend/test_Piece.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    yield pieces
+
+class TestPiece:
+    def test_piece_initialization(self, piece):
+        # Test if the piece is initialized with the expected values
+        assert piece.get_turn() == Turn.WHITE
+        assert piece.get_position() == (-1, -1)
+
+    def test_piece_set_turn(self, piece):
+        piece.set_turn(Turn.BLACK)
```
Comment (EXC3ll3NTrhyTHM) [inline tests/backend/test_Piece.py:23]: From a business logic perspective this should be `set_color`

**Review by EXC3ll3NTrhyTHM — APPROVED** [review]
Looks good

File: tests/backend/test_GameManager.py (hunk 3 above)
Comment (why-xn) [reply tests/backend/test_GameManager.py:26]: I agree the return type should be exception. We should change that in that function.

File: tests/backend/test_Piece.py (hunk 4 above)
Comment (why-xn) [reply tests/backend/test_Piece.py:23]: agreed

**Pull request merged**